File: merge_questions.js
const fs = require('fs');
const path = require('path');

const baseDir = 'c:\\Users\\raffy\\OneDrive\\Desktop\\Programmazione\\test';
const generatedDir = path.join(baseDir, 'Fonti', 'generated_questions');
const questionsPath = path.join(baseDir, 'questions.json');

function mergeQuestions() {
    let allQuestions = [];

    // Load existing questions
    if (fs.existsSync(questionsPath)) {
        try {
            const data = fs.readFileSync(questionsPath, 'utf8');
            allQuestions = JSON.parse(data);
            console.log(`Loaded ${allQuestions.length} existing questions.`);
        } catch (err) {
            console.error('Error reading existing questions:', err);
            allQuestions = [];
        }
    }

    // Load generated batch files
    if (fs.existsSync(generatedDir)) {
        const files = fs.readdirSync(generatedDir);
        let newCount = 0;

        files.forEach(file => {
            if (file.endsWith('.json')) {
                const filePath = path.join(generatedDir, file);
                try {
                    const data = fs.readFileSync(filePath, 'utf8');
                    const batchQuestions = JSON.parse(data);
                    allQuestions = allQuestions.concat(batchQuestions);
                    newCount += batchQuestions.length;
                    console.log(`Loaded ${batchQuestions.length} questions from ${file}`);
                } catch (err) {
                    console.error(`Error reading ${file}:`, err);
                }
            }
        });

        console.log(`Added ${newCount} new questions.`);
    } else {
        console.error('Generated questions directory not found!');
    }

    // Sort by ID
    allQuestions.sort((a, b) => a.id - b.id);

    // Save merged file
    try {
        fs.writeFileSync(questionsPath, JSON.stringify(allQuestions, null, 2), 'utf8');
        console.log(`Successfully saved ${allQuestions.length} questions to questions.json`);
    } catch (err) {
        console.error('Error writing questions.json:', err);
    }
}

mergeQuestions();

File: merge_questions.py
import json
import os
import glob

def merge_questions():
    base_path = r"c:\Users\raffy\OneDrive\Desktop\Programmazione\test"
    questions_path = os.path.join(base_path, "questions.json")
    generated_dir = os.path.join(base_path, "Fonti", "generated_questions")
    
    # Load existing questions
    if os.path.exists(questions_path):
        with open(questions_path, 'r', encoding='utf-8') as f:
            try:
                all_questions = json.load(f)
            except json.JSONDecodeError:
                all_questions = []
    else:
        all_questions = []
    
    print(f"Loaded {len(all_questions)} existing questions.")
    
    # Load and merge batch files
    batch_files = glob.glob(os.path.join(generated_dir, "*_batch_*.json"))
    new_questions_count = 0
    
    for batch_file in batch_files:
        with open(batch_file, 'r', encoding='utf-8') as f:
            try:
                batch_questions = json.load(f)
                all_questions.extend(batch_questions)
                new_questions_count += len(batch_questions)
                print(f"Loaded {len(batch_questions)} questions from {os.path.basename(batch_file)}")
            except json.JSONDecodeError:
                print(f"Error decoding {batch_file}")

    # Sort by ID to be tidy
    all_questions.sort(key=lambda x: x['id'])
    
    # Save back to questions.json
    with open(questions_path, 'w', encoding='utf-8') as f:
        json.dump(all_questions, f, indent=2, ensure_ascii=False)
    
    print(f"Total questions after merge: {len(all_questions)}")
    print(f"Added {new_questions_count} new questions.")

if __name__ == "__main__":
    merge_questions()
